archive/c/c/convex-hull.c

```diff
@@ -0,0 +1,121 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+typedef struct {
+    int x, y;
+} Point;
+
+// Function to determine the orientation of the triplet (p, q, r)
+int orientation(Point p, Point q, Point r) {
+    int val = (q.y - p.y) * (r.x - q.x) - (q.x - p.x) * (r.y - q.y);
+    if (val == 0) return 0;  // Collinear
+    return (val > 0) ? 1 : 2; // Clockwise or counterclockwise
+}
+
+// Function to compute the convex hull using the Gift Wrapping algorithm
+void convexHull(Point points[], int n) {
+    if (n < 3) {
+        printf("Convex hull not possible with less than 3 points.\n");
+        return;
+    }
```

Comment:
Not sure this is actually needed since the number of coordinates was checked before calling this function.